### PR TITLE
Perception: removed unnecessary data fetch from adapter

### PR DESCRIPTION
**Pull request by jinghaomiao**

### CODE REVIEW

File: modules/perception/obstacle/onboard/camera_process_subnode.cc [hunk 1]
```diff
@@ -91,37 +91,35 @@ bool CameraProcessSubnode::InitModules() {
...
 
 void CameraProcessSubnode::ImgCallback(const sensor_msgs::Image &message) {
-  AdapterManager::Observe();
-  sensor_msgs::Image msg = AdapterManager::GetImageFront()->GetLatestObserved();
-
-  double timestamp = msg.header.stamp.toSec();
-
+  double timestamp = message.header.stamp.toSec();
+  ADEBUG << "CameraProcessSubnode ImgCallback: "
+         << " frame: " << ++seq_num_ << " timestamp: ";
```
Comment (eugenegx) [inline modules/perception/obstacle/onboard/camera_process_subnode.cc]: This ++seq_num_ is duplicated. The one below is desired

File: modules/perception/obstacle/onboard/camera_process_subnode.cc [hunk 2]
```diff
@@ -91,37 +91,35 @@ bool CameraProcessSubnode::InitModules() {
...
-            return;
-        }
-        timestamp_ns_ = curr_timestamp;
-      } else {
-        timestamp_ns_ = timestamp * 1e9;
+    if (timestamp_ns_ > 0.0) {
+      double curr_timestamp = timestamp * 1e9;
+
+      if ((curr_timestamp - timestamp_ns_) < (1e9 / FLAGS_camera_hz)) {
+        return;
```
Comment (eugenegx) [inline modules/perception/obstacle/onboard/camera_process_subnode.cc]: I think a ADEBUG here for skipping frame is good, in case later we need it

**Comment by eugenegx** [comment]
Looks good to me! Thank you
The 2 small things I mentioned above are needed

File: modules/perception/obstacle/onboard/camera_process_subnode.cc (hunk 1 above)
Comment (jinghaomiao) [reply modules/perception/obstacle/onboard/camera_process_subnode.cc]: thanks. Fixed.

File: modules/perception/obstacle/onboard/camera_process_subnode.cc (hunk 2 above)
Comment (jinghaomiao) [reply modules/perception/obstacle/onboard/camera_process_subnode.cc]: Thanks. Fixed.

**Pull request merged**